File: SEGUNDO/C2/PDOO/backup/Deep_space/src/deepspace/Weapon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import deepspace.WeaponType;

/**
 *
 * @author daniel
 */
public class Weapon {
    private String name;
    private WeaponType type;
    private int uses;
    
    Weapon(String Name, WeaponType Type, int Uses){
        this.name = Name;
        this.type = Type;
        this.uses = Uses;
    }
    
    Weapon(Weapon s){
        this(s.name, s.type,s.uses);
    }

    public WeaponType getType() {
        return type;
    }

    public int getUses() {
        return uses;
    }
    
    public float power(){
        return type.getPower();
    }
    
    public float useIt(){
        if(uses > 0){
            uses--;
            return 0.0f;
        }
        
        else
            return 1.0f;
    }
    
    WeaponToUI getUIversion(){
        return new WeaponToUI(this);
    }
    
    @Override
    public String toString(){
        String salida="Nombre: " + name;
        
        if(WeaponType.LASER == type)
            salida = salida + "\nTipo: LASER";
        else if (WeaponType.MISSILE == type)
            salida = salida + "\nTipo: MISSILE";
        else
            salida = salida + "\nTipo: PLASMA";
        
        salida= salida +"\nUsos: " + uses + "\n";
        
        return salida;
    }   
}

File: SEGUNDO/C1/NetBeansProjects/DeepSpace_ruby/pkg/DeepSpace_ruby-0.0.1/lib/weapon.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'weapon_type'
require_relative 'WeaponToUI'

module Deepspace
  
class Weapon
  def initialize(name, type, uses)
    @name = name
    @type = type
    @uses = uses
  end
  
  def self.newCopy(copia)
    new(copia.name, copia.type, copia.uses)
  end
  
  attr_reader  :uses
  
  def power
    @type.power
  end
  
  def type 
    if(@type == WeaponType::LASER)
      WeaponType::LASER
    elsif (@type == WeaponType::MISSILE)
      WeaponType::MISSILE
    else
      WeaponType::PLASMA
    end
  end
  
  def useIt
    if(@uses > 0)
      @uses -= 1
      return self.power
    else
      return 1.0
    end
  end
  
  def to_s 
    salida="Nombre: #{@name}\t Usos: #{@uses}\t"
    
    if(@type == WeaponType::LASER)
      salida=salida+ " Tipo: LASER"
    elsif(@type == WeaponType::MISSILE)
      salida = salida + " Tipo: MISSILE"
    else
      salida = salida + " Tipo: PLASMA"
    end
    
    return salida
  end
  
  def getUIVersion()
    WeaponToUI.new(self)
  end

end

end
